### PR TITLE
Error azure client creation if metadata GET fails

### PR DESCRIPTION
**Pull request by PrasadG193**
## Change Overview

Error azure client creation if metadata GET fails

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [x] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

Tested on both types of azure clusters where - metadata was accessible and metadata was not reachable

### CODE REVIEW

**Review by viveksinghggits — APPROVED** [review]
Looks good to me.

**Review by onkarbhat — APPROVED** [review]
LGTM

File: pkg/blockstorage/azure/azure_instance_metadata.go [hunk 1]
```diff
@@ -18,6 +18,8 @@ import (
 	"encoding/json"
 	"io/ioutil"
 	"net/http"
+
+	"github.com/pkg/errors"
```
Comment (tdmanv) [inline pkg/blockstorage/azure/azure_instance_metadata.go:22]: Please use the kanister errors package

Comment (PrasadG193) [reply pkg/blockstorage/azure/azure_instance_metadata.go:22]: @tdmanv I don't think we have Kanister errors package unless I am missing something. We do have Kanister logging package but not errors

Comment (PrasadG193) [reply pkg/blockstorage/azure/azure_instance_metadata.go:22]: I am merging this now, will raise follow up PR based on the discussion

**Pull request merged**